File: lab 7/scala/src/main/scala/Fork.scala
import java.util.concurrent.locks.ReentrantLock


class Fork {
  val lock = new ReentrantLock
  protected var taken = false

  def take(): Unit = {
    taken = true
    lock.lock()
  }

  def putBack(): Unit = {
    lock.unlock()
    taken = false
  }

  def isTaken: Boolean = taken
}

File: lab 7/java/src/Fork.java
import java.util.concurrent.locks.ReentrantLock;

public class Fork
{
    ReentrantLock lock = new ReentrantLock();
    protected boolean taken = false;
    public void take()
    {
        taken = true;
        lock.lock();
    }

    public void putBack()
    {
        lock.unlock();
        taken = false;
    }

    public boolean isTaken()
    {
        return taken;
    }
}
